persist/fs/files.go

```diff
@@ -428,6 +436,49 @@ func IndexSnapshotFiles(filePathPrefix string, namespace ident.ID) (SnapshotFile
 	})
 }
 
+// FileSetAt returns a FileSetFile for the given namespace/shard/blockStart combination if it exists.
+func FileSetAt(filePathPrefix string, namespace ident.ID, shard uint32, blockStart time.Time) (FileSetFile, bool, error) {
+	matched, err := filesetFiles(filePathPrefix, namespace, shard, filesetFilePattern)
+	if err != nil {
+		return FileSetFile{}, false, err
+	}
+
+	matched.sortByTimeAscending()
+	for i, fileset := range matched {
+		if fileset.ID.BlockStart.Equal(blockStart) {
+			nextIdx := i + 1
+			if nextIdx < len(matched) && matched[nextIdx].ID.BlockStart.Equal(blockStart) {
+				// Should never happen
+				return FileSetFile{}, false, fmt.Errorf(
```

Comment:
pull in the invariant utils from https://github.com/m3db/m3x/commit/ef130fab89ea319ba5361d6c73569396bf3c3996 here

Reply:
I pulled in the constant, not sure we want to make an instrument / logger a requirement for using this function

Reply:
It'd be better to emit at a higher layer.